mix/CodeModel.cpp

```diff
@@ -156,14 +161,19 @@ void CodeModel::runCompilationJob(int _jobId, QString const& _code)
 	emit compilationCompleteInternal(result.release());
 }
 
-void CodeModel::onCompilationComplete(CompilationResult*_newResult)
+void CodeModel::onCompilationComplete(CompilationResult* _newResult)
```

Comment:
unique_ptr can't be used as an argument for signal/slot because of a missing copy constructor. In this case I release a pointer from a unique_ptr before emitting it as a signal and immediately capture to unique_ptr upon entering a slot in a new thread. There will be no leak unless there is a serious internal bug in QT signals/slots processing